src/components/app/details/triggerView/cdMaterial.tsx

```diff
@@ -1653,30 +1659,32 @@ const CDMaterial = ({
         }
     }
 
-    const renderTriggerDeployButton = (disableDeployButton: boolean) => (
-        <button
-            data-testid="cd-trigger-deploy-button"
-            disabled={deploymentLoading || isSaveLoading}
-            className={`${getCTAClass(deploymentWindowMetadata.userActionState, disableDeployButton)} h-36`}
-            onClick={(e) => onClickDeploy(e, disableDeployButton)}
-            type="button"
-        >
-            {deploymentLoading || isSaveLoading ? (
-                <Progressing />
-            ) : (
-                <>
-                    {getDeployButtonIcon()}
-                    {deploymentWindowMetadata.userActionState === ACTION_STATE.BLOCKED
-                        ? 'Deployment is blocked'
-                        : CDButtonLabelMap[stageType]}
-                    {isVirtualEnvironment && ' to isolated env'}
-                    {deploymentWindowMetadata.userActionState === ACTION_STATE.BLOCKED && (
-                        <InfoOutline className="icon-dim-16 ml-5" />
-                    )}
-                </>
-            )}
-        </button>
-    )
+    const renderTriggerDeployButton = (disableDeployButton: boolean) => {
+        const userActionState: ACTION_STATE = deploymentWindowMetadata.userActionState
+        if (
+            stageType === DeploymentNodeType.CD &&
+            !disableDeployButton &&
+            (userActionState ? userActionState === ACTION_STATE.ALLOWED : true) &&
+            !(deploymentLoading || isSaveLoading)
+        ) {
+            return <AnimatedDeployButton onButtonClick={onClickDeploy} />
+        }
+        return (
+            <Button
+                dataTestId="cd-trigger-deploy-button"
+                startIcon={getDeployButtonIcon()}
+                isLoading={deploymentLoading || isSaveLoading}
+                text={`${
+                    userActionState === ACTION_STATE.BLOCKED ? 'Deployment is blocked' : CDButtonLabelMap[stageType]
+                }${isVirtualEnvironment ? ' to isolated env' : ''}`}
+                endIcon={userActionState === ACTION_STATE.BLOCKED ? <InfoOutline /> : null}
+                onClick={(e) => onClickDeploy(e, disableDeployButton)}
+                size={ComponentSizeType.large}
+                style={getDeployButtonStyle(userActionState)}
+                disabled={disableDeployButton}
```

Comment:
Will it be disabled in case of deploymentLoading || isSaveLoading

Reply:
<Button /> Component handles disabling on isLoading i.e. deploymentLoading || isSaveLoading in this case